sql3/planner/expression.go

```diff
@@ -41,6 +41,13 @@ func coerceValue(sourceType parser.ExprDataType, targetType parser.ExprDataType,
 				return nil, sql3.NewErrInternalf("unexpected value type '%T'", value)
 			}
 			return pql.NewDecimal(val*int64(math.Pow(10, float64(t.Scale))), t.Scale), nil
+		case *parser.DataTypeTimestamp:
```

Comment:
You may have to change the `typesAreAssignmentCompatible()` function in addition to these changes.

`typesAreAssignmentCompatible()` is called when we are doing an assignment - i.e. expression in an insert list, expression passed as function call parameter, future assignment statement when we have stored code.

Reply:
We could revisit some tests here too in the following areas - not just bulk insert. So something like:

1. ensuring we haven't broken `CAST(1234567 as timestamp)`
2. ensuring `insert into testable (_id, timestampcol) values (1, 1234567)` works as expected
3. ensuring `select datepart('YY', 1234567)` works as expected

Reply:
Tested simple insert and datepart() and cast() functions and found things broken, and made necessary updates in expressiontypes.go and opinsert.go to fix.